src/javascript/app/pages/user/account/authenticate.js

```diff
@@ -482,8 +504,101 @@ const Authenticate = (() => {
+        const mock_response = {
+            needs_verification: ['identity', 'document'],
+            identity          : {
+                status     : 'none',
+                expiry_date: 0,
+            },
+            document: {
+                status     : 'none',
+                expiry_date: 0,
+            },
+        };
+        setTimeout(() => {
+            resolve(mock_response);
+        }, 300);
+    });
+
+    const AuthenticationInit = async () => {
+        // TODO: call the authentication API
+        const authentication_status = await getAuthenticationStatus();
+        if (!authentication_status && authentication_status.error) return;
+        const { identity, document } = authentication_status;
+
+        switch (identity.status) {
+            case 'none':
+                onfidoInit();
+                break;
+            case 'pending':
+                $('#upload_complete').setVisibility(1);
+                break;
+            case 'rejected':
+                $('#unverified').setVisibility(1);
+                break;
+            case 'verified':
+                $('#verified').setVisibility(1);
+                break;
+            default:
+                break;
+        }
+        switch (document.status) {
+            case 'none': {
+                init();
+                $('#not_authenticated').setVisibility(1);
+                const language            = getLanguage();
+                const language_based_link = ['ID', 'RU', 'PT'].includes(language) ? `_${language}` : '';
+                const $not_authenticated  = $('#not_authenticated');
+                let link = Url.urlForCurrentDomain(`https://marketing.binary.com/authentication/Authentication_Process${language_based_link}.pdf`);
+
+                $not_authenticated.setVisibility(1);
+
+                if (Client.isAccountOfType('financial')) {
+                    $('#not_authenticated_financial').setVisibility(1);
+                    link = Url.urlForCurrentDomain('https://marketing.binary.com/authentication/MF_Authentication_Process.pdf');
+                }
+
+                $not_authenticated.find('.learn_more').setVisibility(1).find('a').attr('href', link);
+
+                if (isIdentificationNoExpiry(Client.get('residence'))) {
+                    $('#expiry_datepicker_proofid').setVisibility(0);
+                    $('#exp_date_2').datepicker('setDate', '2099-12-31');
+                }
+                break;
+            }
+            case 'pending':
+                $('#success_message').setVisibility(1);
+                break;
+            case 'rejected':
+                $('#unverified_message').setVisibility(1);
+                break;
+            case 'verified':
+                $('#fully_authenticated').setVisibility(1);
+                break;
+            default:
+                break;
+        }
+        $('#authentication_loading').setVisibility(0);
```

Comment:
ideally loading should be invisible right before making the content visible

Reply:
then i should have two loadings one for content visible, and one to wait for onfido ui to come, since it is async?

Reply:
if it looks fine currently then it's okay

Reply:
yes, it looks fine now